selenium-examples/junit/src/test/java/com/yourcompany/Tests/TestBase.java

```diff
@@ -0,0 +1,75 @@
+package com.yourcompany.Tests;
+
+import com.saucelabs.saucebindings.Browser;
+import com.saucelabs.saucebindings.SauceOptions;
+import com.saucelabs.saucebindings.SaucePlatform;
+import com.saucelabs.saucebindings.SauceSession;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.rules.TestName;
+import org.openqa.selenium.WebDriver;
+
+@Ignore
+public class TestBase {
+    public SauceSession session;
+    protected WebDriver driver;
+
+    @Rule
+    public SauceTestWatcher testWatcher = new SauceTestWatcher();
+
+    @Rule
+    public TestName name = new TestName() {
+        public String getMethodName() {
+            return String.format("%s", super.getMethodName());
+        }
+    };
+
+    @Before
+    public void setUp() {
+        SauceOptions options = new SauceOptions();
+        options.setName(name.getMethodName());
+
+        if (System.getenv("START_TIME") != null) {
+            options.setBuild("Build Time: " + System.getenv("START_TIME"));
+        }
+
+        String platform;
+        if (System.getProperty("platform") != null) {
+            platform = System.getProperty("platform");
+        } else {
+            platform = "default";
+        }
+
+        switch(platform) {
```

Comment:
I have an idea here, would this method be better if we set `platform` and `browser` in Env variables. Then, we would just have a method called 
```
setPlatformAndBrowser()
{
                options.setPlatformName(System.getProperty("platform"));
                options.setBrowserName(System.getProperty("browser"));
}
```
Then we wouldn't have to worry about being limited to this small combination of possibilities? And can remove this ugly switch case.

Reply:
We're starting with the assumption that most of our clients have a lot of tests that they want to run on a smaller number of platforms. So any given code base doesn't need to support an infinite number of configurations, just a way to execute on the 5-10 they care about.

The underlying goal is for the configuration to be in the code not the CI tool. That happens when a CI tool build gets to execute one command `make mac_catalina_safari` or something.

The logic for what all "mac_catalina_safari" does needs to be in the code somewhere. The make file can set a bunch of System Properties for the before hook to process (like you are suggesting), or it can pass along the build command in a system property (like this code) and let the before hook specify the properties. My concern with doing the former in these repos that it is less obvious to the user the pattern that is being followed and they might put all of this configuration into their CI Tool.

In the real world, I would put all of this in a configuration file and the before hook would just auto-load the data. For the purposes of this repo, though, I want to make sure it is obvious to the user what is happening.

Reply:
@titusfortner I think they will turn this into a best practice, as they do with everything we suggest. And future projects will have this logic plus 10 other combinations. I've found that if we don't do it right for 99% of the customers, they won't do it right either. They'll just pick up from our code and run with it.

Reply:
This is the first time this code has been updated in 3 years. We've demonstrated we can't maintain best practices for it. This is better than what has been there, and imo better than using ENV for everything. If they do this it is at least obvious what is happening, and I can easily tweak what they are doing to be better.

We'll point people to best practices once we get something up and running with the framework.